subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -609,13 +602,65 @@ static int mod_member_list_handle(struct bt_mesh_msg_ctx *ctx,
 	return 0;
 }
 
+static int mod_app_list_handle(struct bt_mesh_msg_ctx *ctx, struct net_buf_simple *buf, uint16_t op,
+			       bool vnd)
+{
+	struct mod_member_list_param *param;
+	uint16_t elem_addr, mod_id, cid;
+	uint8_t status;
+
+	status = net_buf_simple_pull_u8(buf);
```

Comment:
Need length checking here too. Malformed frame shouldn't crash device.

Reply:
I still think when optional fields present, then length should be checked since this is differ to `BT_MESH_LEN_MIN(N)`.

Reply:
I didn't get the question. Trust ourselves in memory analyzing behind allocated buffer?

Reply:
When `OP_MOD_SUB_LIST_VND` opcode is used, the min length is 7, when opcode is `OP_MOD_SUB_LIST`, length is 5. This is guaranteed by `BT_MESH_LEN_MIN` macro. The `OP_MOD_SUB_LIST` opcode won't call 
 `mod_sub_list_vnd` handler. So why do we need additional check?